pkg/watermark/fetch/edge_fetcher_set.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The Numaproj Authors.
```

Comment:
nit: shall we update all Copyright to 2023? Derek has a script under the `hack` folder to do so.